python/morphology.cpp

```diff
@@ -24,10 +36,178 @@ void register_morphology(pybind11::module& m) {
+            })
+        .def("__repr__",
+            [](arb::mlocation l) {
+                return util::pprintf("<arbor.location: branch {}, position {}>", l.branch, l.pos);
+            });
+
+    // arb::mpoint
+    pybind11::class_<arb::mpoint> mpoint(m, "mpoint");
+    mpoint
+        .def(pybind11::init(
+                [](double x, double y, double z, double r) {
+                    return arb::mpoint{x,y,z,r};
+                }),
+                "x"_a, "y"_a, "z"_a, "radius"_a, "All values in μm.")
+        .def_readonly("x", &arb::mpoint::x, "X coordinate [μm].")
+        .def_readonly("y", &arb::mpoint::y, "Y coordinate [μm].")
+        .def_readonly("z", &arb::mpoint::z, "Z coordinate [μm].")
+        .def_readonly("radius", &arb::mpoint::radius,
+            "Radius of cable at sample location centered at coordinates [μm].")
+        .def("__str__",
+            [](const arb::mpoint& p) {
+                return util::pprintf("<arbor.mpoint: x {}, y {}, z {}, radius {}>", p.x, p.y, p.z, p.radius);
+            })
+        .def("__repr__",
+            [](const arb::mpoint& p) {return util::pprintf("{}>", p);});
 
+    // arb::msample
+    pybind11::class_<arb::msample> msample(m, "msample");
+    msample
+        .def(pybind11::init(
+            [](arb::mpoint loc, int tag){
+                return arb::msample{loc, tag};}),
+            "location"_a, "tag"_a)
+        .def(pybind11::init(
+            [](double x, double y, double z, double r, int tag){
+                return arb::msample{{x,y,z,r}, tag};}),
+            "x"_a, "y"_a, "z"_a, "radius"_a, "tag"_a,
+            "spatial values {x, y, z, radius} are in μm.")
+        .def_readonly("loc", &arb::msample::loc,
+            "Location of sample.")
+        .def_readonly("tag", &arb::msample::tag,
+            "Property tag of sample. "
+            "If loaded from standard SWC file the following tags are used: soma=1, axon=2, dendrite=3, apical dendrite=4, however arbitrary tags can be used.")
+        .def("__str__",
+            [](const arb::msample& s) {
+                return util::pprintf("<arbor.mpoint: x {}, y {}, z {}, radius {}, tag {}>",
+                        s.loc.x, s.loc.y, s.loc.z, s.loc.radius, s.tag); })
+        .def("__repr__",
+            [](const arb::msample& s) {
+                return util::pprintf("{}>", s);});
+
+    // arb::mcable
+    pybind11::class_<arb::mcable> cable(m, "cable");
+    cable
+        .def(pybind11::init(
+                    [](arb::msize_t bid, double prox, double dist) {
+                        arb::mcable c{bid, prox, dist};
+                        if (!test_invariants(c)) {
+                            throw pyarb_error("Invalid cable description. Cable segments must have proximal and distal end points in the range [0,1].");
+                        }
+                        return c;
+                    }),
+             "branch_id"_a, "prox"_a, "dist"_a)
```

Comment:
add description, e.g. `Create an unbranched cable in the morphology with the id of the branch on which the cable lies, the relative position of the proximal end of the cable on its branch and the relative position of the distal end of the cable on its branch.`